app/models/section.rb

```diff
@@ -118,8 +118,9 @@ def duplicate(helper=nil)
           embed.primary_linked_items.each do |pli|
             primary = helper.get_copy(embed)
             secondary = helper.get_copy(pli.secondary.embeddable)
-            if primary && secondary
-              lpi = LinkedPageItem.new(primary_id: primary.page_item.id, secondary_id: secondary.page_item.id, label: pli.label)
+            secondary_page_item = secondary.page_item ? secondary.page_item : pli.secondary.embeddable.page_item
```

Comment:
A comment here would be useful. Something like:
"If the secondary embeddable is in a different section, it won't be copied. So in this case the new primary embeddable references the original secondary embeddable."